### PR TITLE
speed up under ThreadSanitizer

### PR DESCRIPTION
**Pull request by bejado**

### CODE REVIEW

File: libs/utils/include/utils/compiler.h [hunk 1]
```diff
@@ -33,7 +33,7 @@
 #if __has_attribute(visibility)
 #    define UTILS_PUBLIC  __attribute__((visibility("default")))
 #else
-#    define UTILS_PUBLIC
```
Comment (romainguy) [inline libs/utils/include/utils/compiler.h:36]: This doesn't match the indent above

File: libs/utils/include/utils/compiler.h [hunk 2]
```diff
@@ -65,19 +65,21 @@
 #endif
 
 #define UTILS_NO_SANITIZE_THREAD
-#if defined(__has_feature)
-#    if __has_feature(thread_sanitizer)
-#        undef UTILS_NO_SANITIZE_THREAD
-#        define UTILS_NO_SANITIZE_THREAD __attribute__((no_sanitize("thread")))
-#    endif
+#if __has_feature(thread_sanitizer)
```
Comment (bejado) [inline libs/utils/include/utils/compiler.h:68]: I think we still need to check for `defined(__has_feature)`, otherwise MSVC complains.

Comment (dvyukov) [reply libs/utils/include/utils/compiler.h:68]: There is:
```
#ifndef __has_feature
#define __has_feature(x) 0
#endif
```
at the top of this file and existing uses of __has_feature w/o the check, e.g.:
```
#elif __has_feature(cxx_thread_local)
```

Comment (bejado) [reply libs/utils/include/utils/compiler.h:68]: Ah thanks I see that now.

**Review by pixelflinger — APPROVED** [review]
the gcc/clang one is a very good catch!

**Pull request merged**